### PR TITLE
Add thrift benchmark file

### PR DESCRIPTION
**Pull request by willhug**

### CODE REVIEW

File: encoding/thrift/benchmark_test.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import (
+	"bytes"
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"go.uber.org/thriftrw/wire"
+	"go.uber.org/yarpc/api/transport"
```
Comment (abhinav) [inline encoding/thrift/benchmark_test.go]: import groups

File: encoding/thrift/benchmark_test.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	proto := NewMockProtocol(mockCtrl)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	proto.EXPECT().DecodeEnveloped(gomock.Any()).Return(wire.Envelope{
+		Name:  "someMethod",
+		SeqID: 42,
+		Type:  wire.Exception,
+		Value: requestBody,
```
Comment (abhinav) [inline encoding/thrift/benchmark_test.go]: Is the cost of the mock protocol not an issue?

Comment (willhug) [inline encoding/thrift/benchmark_test.go]: not in terms of relative benchmarking

Comment (willhug) [inline encoding/thrift/benchmark_test.go]: possibly this should be better thought out as an end-to-end test, but I was trying to focus on a few key points.

**Comment by abhinav** [comment]
Please update PR title and description

File: encoding/thrift/benchmark_test.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	"time"
+
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/api/transport/transporttest"
+
+	"github.com/golang/mock/gomock"
+	"go.uber.org/thriftrw/wire"
+)
+
+func BenchmarkThrift(b *testing.B) {
```
Comment (bombela) [inline encoding/thrift/benchmark_test.go]: are you bench-marking the initialization of the test? if not, testing.B offers a way to reset the timer before running the b.N loop.

File: encoding/thrift/benchmark_test.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	mockCtrl := gomock.NewController(b)
+	defer mockCtrl.Finish()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	proto := NewMockProtocol(mockCtrl)
+	proto.EXPECT().DecodeEnveloped(gomock.Any()).Return(wire.Envelope{
```
Comment (bombela) [inline encoding/thrift/benchmark_test.go]: you are benchmarking the generated mock code at this point. Maybe you only need to call encode/decode in the loop.

**Comment by breerly** [comment]
@willhug can you either close this or update the PR description with background/motivation and land?

**Comment by willhug** [comment]
We should probably re-evalutate this with a better benchmark suite, and trying to get it landed has been a pain.

**Pull request closed without merging**